dockerfiles/cli/scripts/cli.sh

```diff
@@ -86,19 +86,65 @@ server_is_booted_extra_check() {
 }
 
 cmd_init_reinit_pre_action() {
-    sed -i'.bak' "s|#CODENVY_HOST=.*|CODENVY_HOST=${CODENVY_HOST}|" "${REFERENCE_CONTAINER_ENVIRONMENT_FILE}"
-    sed -i'.bak' "s|#CODENVY_SWARM_NODES=.*|CODENVY_SWARM_NODES=${CODENVY_HOST}:23750|" "${REFERENCE_CONTAINER_ENVIRONMENT_FILE}"
+  sed -i'.bak' "s|#CODENVY_HOST=.*|CODENVY_HOST=${CODENVY_HOST}|" "${REFERENCE_CONTAINER_ENVIRONMENT_FILE}"
+  sed -i'.bak' "s|#CODENVY_SWARM_NODES=.*|CODENVY_SWARM_NODES=${CODENVY_HOST}:23750|" "${REFERENCE_CONTAINER_ENVIRONMENT_FILE}"
+
+  # For testing purposes only
+  #HTTP_PROXY=8.8.8.8
+  #HTTPS_PROXY=http://4.4.4.4:9090
+  #NO_PROXY="locahost, *.local, swarm-mode"
+
+  if [[ ! ${HTTP_PROXY} = "" ]]; then
+    sed -i'.bak' "s|#${CHE_PRODUCT_NAME}_HTTP_PROXY_FOR_${CHE_PRODUCT_NAME}=.*|${CHE_PRODUCT_NAME}_HTTP_PROXY_FOR_${CHE_PRODUCT_NAME}=${HTTP_PROXY}|" "${REFERENCE_CONTAINER_ENVIRONMENT_FILE}"
+    sed -i'.bak' "s|#${CHE_PRODUCT_NAME}_HTTP_PROXY_FOR_${CHE_PRODUCT_NAME}_WORKSPACES=.*|${CHE_PRODUCT_NAME}_HTTP_PROXY_FOR_${CHE_PRODUCT_NAME}_WORKSPACES=${HTTP_PROXY}|" "${REFERENCE_CONTAINER_ENVIRONMENT_FILE}"
+  fi
+  if [[ ! ${HTTPS_PROXY} = "" ]]; then
+    sed -i'.bak' "s|#${CHE_PRODUCT_NAME}_HTTPS_PROXY_FOR_${CHE_PRODUCT_NAME}=.*|${CHE_PRODUCT_NAME}_HTTPS_PROXY_FOR_${CHE_PRODUCT_NAME}=${HTTPS_PROXY}|" "${REFERENCE_CONTAINER_ENVIRONMENT_FILE}"
+    sed -i'.bak' "s|#${CHE_PRODUCT_NAME}_HTTPS_PROXY_FOR_${CHE_PRODUCT_NAME}_WORKSPACES=.*|${CHE_PRODUCT_NAME}_HTTPS_PROXY_FOR_${CHE_PRODUCT_NAME}_WORKSPACES=${HTTPS_PROXY}|" "${REFERENCE_CONTAINER_ENVIRONMENT_FILE}"
+  fi
+  if [[ ! ${NO_PROXY} = "" ]]; then
+    #
+    # NOTE --- Notice that if no proxy is set, we must append 'codenvy-swarm' to this for docker networking
+    #
+    sed -i'.bak' "s|#${CHE_PRODUCT_NAME}_NO_PROXY_FOR_${CHE_PRODUCT_NAME}=.*|${CHE_PRODUCT_NAME}_NO_PROXY_FOR_${CHE_PRODUCT_NAME}=${NO_PROXY},codenvy-swarm|" "${REFERENCE_CONTAINER_ENVIRONMENT_FILE}"
```

Comment:
If we add the `codenvy-swarm` host at the end of the `NO_PROXY` string, it must be followed by comma (because of non-stansdart hostname), or insert it into beginning of the string or between IP's. Examples:
Not working:
```
CODENVY_NO_PROXY_FOR_CODENVY=localhost,127.0.0.1,172.17.0.1,codenvy-swarm
```
Working:
```
CODENVY_NO_PROXY_FOR_CODENVY=localhost,127.0.0.1,172.17.0.1,codenvy-swarm,
```
or
```
CODENVY_NO_PROXY_FOR_CODENVY=localhost,127.0.0.1,codenvy-swarm,172.17.0.1
```
or
```
CODENVY_NO_PROXY_FOR_CODENVY=codenvy-swarm,localhost,127.0.0.1,172.17.0.1
```